src/sync.rs

```diff
@@ -1,7 +1,30 @@
-//! A write-once cell mediated by the Python GIL.
+//! Synchronization mechanisms based on the Python GIL.
 use crate::{types::PyString, Py, Python};
 use std::cell::UnsafeCell;
 
+/// Value with concurrent access protected by the GIL.
+///
+/// This is a synchronization primitive based on Python's global interpreter lock (GIL).
+/// It ensures that only one thread at a time can access the inner value via shared references.
+/// It can be combined with interior mutability to obtain mutable references.
+pub struct GILProtected<T> {
```

Comment:
This should have a doc example. Something with a `static `GILProtected<RefCell<T>>`, I'd imagine.

Reply:
Added an example. Do you think this is sufficient or would should it be more involved, e.g. using multiple threads or asserting on the result of the modification?

Reply:
Lovely, thanks.